commands/guest.js

```diff
@@ -0,0 +1,490 @@
+    console.log(`Remove timeout: ${snowflake}`);
+    timeoutMap.delete(snowflake);
+    return true;
+  }
+  console.log(`Could not find key ${snowflake}`);
+  return false;
+};
+
+/**
+ * Removes the timeout for a user with a given snowflake, if one exists, and
+ * processes the rest of the timeouts.
+ * @param {string} snowflake The snowflake of the user to remove a timeout for.
+ * @return {bool} true if a timeout was removed, false otherwise.
+ */
+function removeTimeoutAndUpdate(snowflake) {
+  if (removeTimeout(snowflake)) {
+    processTimeouts();
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Remove a user from the server. The user will not be removed and their
+ * timeout will be cleared if they do not have the guest role. The timeout will
+ * be extended for 1 hour if the user is in a voice channel at the time the
+ * the timeout expires.
+ * @param {TimeoutEntry} userEntry The TimeoutEntry containing the user and
+ * timeout information.
+ */
+function removeUser(userEntry) {
+  if (!userEntry.user.roles.has(guestRole)) {
+    console.log(`User is not guest: ${userEntry.toEntryString()}`);
+    // TODO(aalberg) Remove this debug statement when we're confident
+    // everything is working correctly.
+    const channel = util.getChannelById(discordClient, debugChannelId);
+    if (util.exists(channel)) {
+      channel.send(`User is not guest: ${userEntry.toEntryString()}`);
+    }
+    removeTimeout(userEntry.user.id);
+    return;
+  }
+  console.log(`Voice channel ${userEntry.user.voiceChannel}`);
+  if (util.exists(userEntry.user.voiceChannel)) {
+    console.log(`Delaying kick of: ${userEntry.toEntryString()}`);
+    // TODO(aalberg) Remove this debug statement when we're confident
+    // everything is working correctly.
+    const channel = util.getChannelById(discordClient, debugChannelId);
+    if (util.exists(channel)) {
+      channel.send(`Delaying kick of: ${userEntry.toEntryString()}`);
+    }
+    addTimeout(userEntry.user, util.addHours(new Date(), 1), true);
+  } else {
+    console.log(`Kicking: ${userEntry.toEntryString()}`);
+    // TODO(aalberg): Update this to actually remove people when we're confident
+    // everything is working correctly.
+    const channel = util.getChannelById(discordClient, debugChannelId);
+    if (util.exists(channel)) {
+      channel.send(`Kicking: ${userEntry.toEntryString()}`);
+    }
+    removeTimeout(userEntry.user.id);
+  }
+}
+
+/**
+ * Sets the wakeup time of the cron job to the specified time and starts the
+ * cron job.
+ * @param {Date} time The wakeup time of the cron job.
+ */
+function setCronTimeout(time) {
+  console.log('New cron time: ' + time);
+  if (!util.exists(timeoutCron)) {
+    timeoutCron = cron.job(time, processTimeouts);
+  } else {
+    timeoutCron.setTime(cron.time(time));
+  }
+  timeoutCron.start();
+}
+
+/**
+ * Process all of the currently active timeouts and remove guests whos timeouts
+ * have expired.
+ */
+function processTimeouts() {
```

Comment:
Should we call this updateTimeouts (or the ones above xTimeoutAndProcess) for consistency?